.clang-tidy

```diff
@@ -2,29 +2,16 @@
 Checks: "bugprone-use-after-move,\
 clang-analyzer-*,\
 clang-diagnostic-*,\
-google-*,\
+google-objc-*,\
+google-explicit-constructor \
 modernize-use-default-member-init,\
 readability-identifier-naming,\
 -google-objc-global-variable-declaration,\
 -google-objc-avoid-throwing-exception,\
 -clang-analyzer-nullability.NullPassedToNonnull,\
 -clang-analyzer-nullability.NullablePassedToNonnull,\
 -clang-analyzer-nullability.NullReturnedFromNonnull,\
--clang-analyzer-nullability.NullableReturnedFromNonnull,\
-performance-move-const-arg,\
-performance-unnecessary-value-param"
```

Comment:
Turn off `performance-unnecessary-value-param`, all platforms had failures:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799824923980615249/+/u/test:_lint_ios_debug_sim/stdout
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799824923980615265/+/u/test:_lint_host_debug/stdout
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799824923350402209/+/u/test:_lint_host_debug/stdout
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799824923350402193/+/u/test:_lint_android_debug_arm64/stdout

Also added in #36221

Reply:
cc @gaaclarke

Reply:
Right, the intention was to keep that linter in the warning phase.  It requires quite a bit of effort to migrate the code since the clang tidy fix is imperfect for this one.  In order to run the clang tidy fix you need to be building the engine on your machine and takes like 30 minutes to execute.  So I would like to not slip on this.  Is this related to the execution time of clang-tidy?  I'm fine removing this lint to be something that isn't run on CI but I'd like it showing up for developers locally so that we don't introduce problems.

Reply:
I think we should keep these, I've turned them on as errors in https://github.com/flutter/engine/pull/36910